app/assets/javascripts/index/directions.js

```diff
@@ -142,12 +142,20 @@ OSM.Directions = function (map) {
 
   $(".directions_form .reverse_directions").on("click", function () {
     var from = endpoints[0].latlng,
-        to = endpoints[1].latlng;
+        to = endpoints[1].latlng,
+        routeFrom = "",
+        routeTo = "";
+    if (from) {
+      routeFrom = from.lat + "," + from.lng;
+    }
+    if (to) {
+      routeTo = to.lat + "," + to.lng;
+    }
 
     OSM.router.route("/directions?" + querystring.stringify({
       from: $("#route_to").val(),
       to: $("#route_from").val(),
```

Comment:
Note that THESE `to` things is the property for the object of the `stringyfy` and not the variable six lines above

Reply:
We probably should rename the variables if they are too confusing here. But I wanted a minimal fix in the first place.